Readme fixes (#5)
CodeforAustralia/civic-makers-dev-onboarding: README.md

## Asking technical questions:

Take your time and describe as much of the context as you can manage. You certainly don’t have to assume zero knowledge, but it is really important to think about how you can make it as clear as possible to someone who might not have all the context but may still be able to help. Taking some time to think about how best to ask the question will help you to get the best sort of responses. It’s much more important than you might think, and small efforts to be thoughtful here have big flow-on effects.

## Code style:

> Keep it simple. I sound really boring saying this, but we just want nice simple code that works and that people can understand. Obviously this is within reason, and is also somewhat subjective. But the first thing to consider when writing the code is how easily it will be understood, and how will it make things easier to be build upon. Remember that this code will need to be read by other developers of different skill levels, so just writing code that works is really the baseline.  

Reviewer: Hi @batteries76 ,
I think we it comes to explaining code style, it's best accompanied by concrete examples.

Even better - 'codify' the expectations in the form of a (set of?) linter configs, and there will be no ambiguities regarding what 'our code style' is!

Author: Yeah, sorry, can probably break that into two categories. 

One is the linting side of things, which we can chat about and implement. And I can add that to the document.

The other is more amorphous, and harder to codify. 
What I'm trying to avoid is complicated or 'clever' code that is hard for other people to understand or read. The sort of thing where we would like to spend three lines instead of one so that we can see what's going on. Not sure how to codify it - although I can probably find some examples:
```
processedData = await res.json().split(" ").filter(/s/).join("/").pop().etc()
```
That isn't real code, just a codish blurb to demonstrate. 
Should be at least a few lines that can help people see the transition to the final data with sensible variable names. I can add some of this into the document.

Reviewer: Hi @batteries76 ,

thx for looking into the examples :bow: 
I agree setting good linting configurations/standards doesn't get us all the way there in terms of ensuring good 'code style'.

But I think the linting warnings it will give ppl quicker feedback about their code quality, which is why I think it's worth mentioning. And it's somewhat unbiased too ; either the lint checks fail, or it passes. For example, setting a maximum line length in the linter config might have caught the `processedData` example you mentioned.

Also, we want to be mindful not to make this readme lengthier than what it needs to be too! There's plenty that can be said about 'good code style', but this readme is not a blog about it! I suggest that it should be more than sufficient to offer a handful of useful links for the reader to browse through about this vast topic.

> Keep it simple. I sound really boring saying this, but we just want nice simple code that works and that people can understand. Obviously this is within reason, and is also somewhat subjective. But the first thing to consider when writing the code is how easily it will be understood, and how will it make things easier to be build upon. Remember that this code will need to be read by other developers of different skill levels, so just writing code that works is really the baseline.  

Reviewer: >  I sound really boring saying this, but ....

Some of the sentences are in first person perspective; maybe it might be better to phrase it differently?

```suggestion
Keep it simple. It sounds really boring saying this, but ....
```

Author: Great point.

## Trello to PR:

    - Take a Trello task: assign it to yourself, add any notes as needed. 
    - You can keep this simple, but if you want to communicate please do - we welcome this, and it makes other people’s lives easier. You can mention anything about this task that you might want to here. Please read the How We Use Trello reference here: https://www.notion.so/Civic-Makers-ba31d86ca9624c01b9569906c5d692f9?p=a040d9930e8944948b5eff65065941c5

>     - You can reference this Trello task in your PR (see the next section).

Reviewer: Please remove the indentation in each bullet point.

```suggestion
- Take a Trello task: assign it to yourself, add any notes as needed. 
- You can keep this simple, but if you want to communicate please do - we welcome this, and it makes other people’s lives easier. You can mention anything about this task that you might want to here. Please read the [How We Use Trello reference doc](https://www.notion.so/Civic-Makers-ba31d86ca9624c01b9569906c5d692f9?p=a040d9930e8944948b5eff65065941c5)
- You can reference this Trello task in your PR (see the next section).
```

Author: Done.

The system:

>     1. You should fork the project from the original repo.
>     2. Get it running locally.
>     3. For new work create a new branch with an appropriate name.
>     4. Make changes, and commit.
>     5. Push the changes to your repo.
>     6. Make a PR from your repo to the original.
>     7. Respond to correspondence about the PR (if needed).
>     8. Make changes (if needed).
>     9. And on you go.. 

Reviewer: Please remove the indentation in each bullet point.

```suggestion
1. You should fork the project from the original repo.
2. Get it running locally.
3. For new work create a new branch with an appropriate name.
4. Make changes, and commit.
5. Push the changes to your repo.
6. Make a PR from your repo to the original.
7. Respond to correspondence about the PR (if needed).
8. Make changes (if needed).
9. And on you go.. 
```

Author: Done.

A few other pointers:

>     - Meaningful PR messages. These should be roughly aligned with the significance of the PR. For anything complex, be prepared to thoroughly explain in the PR what the aim of the PR is, and how you achieve that aim. Again, this is within reason, but for particularly thorny parts of the code, please explain anything that might need it. 
>     - Add any tags to the PR that are needed (WIP, Spike, etc)
>     - If the PR is from a Trello task, please add a link to that task.
>     - If you make some git errors, don’t hesitate to have a chat with someone from the team.

Reviewer: Please remove the indentation in each bullet point.

```suggestion
- Meaningful PR messages. These should be roughly aligned with the significance of the PR. For anything complex, be prepared to thoroughly explain in the PR what the aim of the PR is, and how you achieve that aim. Again, this is within reason, but for particularly thorny parts of the code, please explain anything that might need it. 
- Add any tags to the PR that are needed (WIP, Spike, etc)
- If the PR is from a Trello task, please add a link to that task.
- If you make some git errors, don’t hesitate to have a chat with someone from the team.
```

Author: Done.

> ## Pair programming:

Reviewer: Are there any useful links to Pair / Mob programming that we could link to in this section?

Author: Yeah, great idea. I'll have a look.

## How to help more generally:

>     - Even if junior, you can read PRs. This is also work. Even if it’s just to say good job, or that you don’t understand something and could it be made clearer. Reading and responding to PRs is a skill, and one that I’m still learning. Dive in, or if you are worried about taking steps, ask someone to give you a hand. 
>     - We love love love people who contribute to the README. If you document your struggles as you go through, these can be excellent elements to add to the README or form the basis for a FAQ. This helps others in a big way.
>     - Get involved on Slack where you can.

Reviewer: Please remove the indentation in each bullet point.

```suggestion
- Even if junior, you can read PRs. This is also work. Even if it’s just to say good job, or that you don’t understand something and could it be made clearer. Reading and responding to PRs is a skill, and one that we're still learning. Dive in, or if you are worried about taking steps, ask someone to give you a hand. 
- We love love love people who contribute to the README. If you document your struggles as you go through, these can be excellent elements to add to the README or form the basis for a FAQ. This helps others in a big way.
- Get involved on Slack where you can.
```

Author: Done.
